client/editorQueryEvaluator.ts

```diff
@@ -0,0 +1,221 @@
+'use strict';
+import { ExtensionContext, TextDocument, TextEdit, TextEditor, Uri, WorkspaceEdit,
+    commands, window, workspace, WorkspaceConfiguration } from 'vscode';
+import { cascadeOverrideClient } from './vscQueryParameterTools';
+import { ClientContext, sendJSQuery, sendSparql, sendXQuery, sendRows } from './marklogicClient';
+import { ClientResponseProvider } from './clientResponseProvider';
+import { contentType, RowsResponse } from 'marklogic';
+import * as ml from 'marklogic';
+
+export enum EditorQueryType {
+    JS,
+    XQY,
+    SPARQL,
+    SQL,
+    ROWS
+}
+
+export class EditorQueryEvaluator {
+    private static FOPTIONS = { tabSize: 2, insertSpaces: true };
+    private static FCOMMAND = 'vscode.executeFormatDocumentProvider';
+    private static SJS = 'sjs';
+    private static XQY = 'xqy';
+
+    private context: ExtensionContext;
+    private provider: ClientResponseProvider;
+
+    public constructor(context: ExtensionContext, provider: ClientResponseProvider) {
+        this.context = context;
+        this.provider = provider;
+    }
+
+    // Exported only for testing purposes
+    public static getQueryTextFromEditor(editor: TextEditor): string {
+        const selection = editor.selection;
+        const selectionText = editor.document.getText(selection);
+        let actualQuery: string = null;
```

Comment:
I think we can just call this `query` - the use of `actual` makes me think that mlxprs took the user's query (whether it was the full editor text or only highlighted text) and then altered it in some fashion. But it's just a `query`, no need for any qualifier on it. And I think the method can be named `getQueryFromEditor`, and this method has the smarts to know whether the query is all of the text, or only the selected text.